libraries/botbuilder-dialogs-adaptive-runtime/src/coreBotAdapter.ts

```diff
@@ -1,50 +1,76 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
-import { ClaimsIdentity, SkillValidation } from 'botframework-connector';
+import * as t from 'runtypes';
+import { BotFrameworkAuthentication, ClaimsIdentity, SkillValidation } from 'botframework-connector';
 
 import {
     ActivityEx,
-    BotFrameworkAdapter,
-    BotFrameworkAdapterSettings,
+    CloudAdapter,
     ConversationState,
     InputHints,
     MessageFactory,
+    Response,
     TurnContext,
     UserState,
     useBotState,
 } from 'botbuilder';
 
-export class CoreBotAdapter extends BotFrameworkAdapter {
+const ErrorT = t.Record({
+    message: t.String,
+    statusCode: t.Optional(t.Number),
+});
+
+export class CoreBotAdapter extends CloudAdapter {
     constructor(
-        settings: Partial<BotFrameworkAdapterSettings>,
+        botFrameworkAuthentication: BotFrameworkAuthentication,
         private readonly conversationState: ConversationState,
         userState: UserState
     ) {
-        super(settings);
+        super(botFrameworkAuthentication);
 
         useBotState(this, userState, conversationState);
 
-        this.onTurnError = async (turnContext, err) => {
+        this.onTurnError = async (context, err) => {
             console.error('[onTurnError] unhandled error', err);
 
-            await this.sendErrorMessage(turnContext, err);
-            await this.sendEoCToParentIfSkill(turnContext, err);
-            await this.clearConversationState(turnContext);
+            const res = context.turnState.get<Response | undefined>(this.ResponseKey);
+
+            if (res) {
```

Comment:
I'm open to suggestions here - we don't really want to re-throw the exception out of `onTurnError`, so in order to reply to the HTTP request that initiated execution, we must handle it here.

Reply:
Ended up going with that approach.